src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java

```diff
@@ -0,0 +1,397 @@
+		logger.info("Get systems request started...");
+		SystemListResponseDTO response = arrowheadService.consumeServiceHTTP(SystemListResponseDTO.class,
+				HttpMethod.GET,
+				Utilities.createURI(getScheme(),
+						CommonConstants.$SERVICEREGISTRY_ADDRESS_WD.split(":")[1].replace("}", ""),
+						Integer.parseInt(CommonConstants.$SERVICEREGISTRY_PORT_WD.split(":")[1].replace("}", "")),
+						CommonConstants.SERVICEREGISTRY_URI + ConsumerConstants.QUERY_GET_SYSTEMS),
+				null, null);
+		return response.getData();
+	}
+
+	// -------------------------------------------------------------------------------------------------
+	private List<ServiceDefinitionResponseDTO> getServices() {
+		logger.info("Get services request started...");
+		ServiceDefinitionsListResponseDTO response = arrowheadService.consumeServiceHTTP(
+				ServiceDefinitionsListResponseDTO.class, HttpMethod.GET,
+				Utilities.createURI(getScheme(),
+						CommonConstants.$SERVICEREGISTRY_ADDRESS_WD.split(":")[1].replace("}", ""),
+						Integer.parseInt(CommonConstants.$SERVICEREGISTRY_PORT_WD.split(":")[1].replace("}", "")),
+						CommonConstants.SERVICEREGISTRY_URI + ConsumerConstants.QUERY_GET_SERVICES),
+				null, null);
+		return response.getData();
+	}
+
+	// -------------------------------------------------------------------------------------------------
+	private List<ServiceInterfaceResponseDTO> getInterfaces() {
+		logger.info("Get interfaces request started...");
+		ServiceInterfacesListResponseDTO response = arrowheadService.consumeServiceHTTP(
+				ServiceInterfacesListResponseDTO.class, HttpMethod.GET,
+				Utilities.createURI(getScheme(),
+						CommonConstants.$SERVICEREGISTRY_ADDRESS_WD.split(":")[1].replace("}", ""),
+						Integer.parseInt(CommonConstants.$SERVICEREGISTRY_PORT_WD.split(":")[1].replace("}", "")),
+						CommonConstants.SERVICEREGISTRY_URI + ConsumerConstants.QUERY_GET_INTERFACES),
+				null, null);
+		return response.getData();
+	}
+
+	// -------------------------------------------------------------------------------------------------
+	private List<AuthorizationIntraCloudResponseDTO> getAuthorizationRules() {
+		Map<String, String> authorizationUri = null;
+		List<AuthorizationIntraCloudResponseDTO> rules = null;
+		try {
+			authorizationUri = getAuthorizationUri();
+		} catch (Exception e) {
+			logger.error(e.getMessage());
+		}
+		logger.info("Get authorization rules request started...");
+		AuthorizationIntraCloudListResponseDTO response = arrowheadService.consumeServiceHTTP(
+				AuthorizationIntraCloudListResponseDTO.class, HttpMethod.GET,
+				Utilities.createURI(authorizationUri.get("scheme"), authorizationUri.get("host"),
+						Integer.parseInt(authorizationUri.get("port")),
+						authorizationUri.get("path") + ConsumerConstants.QUERY_AUTH_INTRA_CLOUD),
+				null, null);
+
+		if (response == null) {
+			logger.error("Getting the authorization rules was unsuccessful!");
+		} else if (response.getData().isEmpty()) {
```

Comment:
I am not sure it is a problem if there are no current authorization rules. It just means you don't have to delete anything before you add the new ones.